core/conversion/var/Var.cpp

```diff
@@ -90,24 +90,37 @@ nvinfer1::ITensor* Var::ITensorOrFreeze(ConversionCtx* ctx) {
     LOG_DEBUG(ctx->logger, "Found IValue containing object of type " << *(ptr_.ivalue->type()));
   }
   TRTORCH_CHECK(
-      isITensor() || (isIValue() && ptr_.ivalue->isTensor()),
+      isITensor() || (isIValue() && (ptr_.ivalue->isTensor() || ptr_.ivalue->isCustomClass())),
       "Requested either IValue containing a Tensor, or ITensor, however Var type is " << type_name());
 
   nvinfer1::ITensor* out;
-
+  auto weights = converters::Weights();
   if (isIValue()) {
-    auto weights = converters::Weights(ctx, ptr_.ivalue->toTensor());
-
-    auto const_layer = ctx->net->addConstant(weights.shape, weights.data);
-    TRTORCH_CHECK(const_layer, "Unable to freeze tensor into constant layer");
-
-    out = const_layer->getOutput(0);
-
-    std::ostringstream tensor_id;
-    tensor_id << reinterpret_cast<int*>(out);
-
-    LOG_DEBUG(ctx->logger, "Freezing tensor " << tensor_id.str() << " as an IConstantLayer");
-    const_layer->setName(("[Freeze Tensor " + tensor_id.str() + " ]").c_str());
+    if (ptr_.ivalue->isTensor()) {
+      auto tensor = ptr_.ivalue->toTensor();
+      if (tensor.scalar_type() == at::kLong) {
+        weights = converters::Weights(ctx, tensor.toType(at::kInt));
+      } else if (tensor.scalar_type() == at::kDouble) {
+        weights = converters::Weights(ctx, tensor.toType(at::kFloat));
+      } else {
+        weights = converters::Weights(ctx, tensor);
+      }
+
+      auto const_layer = ctx->net->addConstant(weights.shape, weights.data);
+      TRTORCH_CHECK(const_layer, "Unable to freeze tensor into constant layer");
+
+      out = const_layer->getOutput(0);
+
+      std::ostringstream tensor_id;
+      tensor_id << reinterpret_cast<int*>(out);
+
+      LOG_DEBUG(ctx->logger, "Freezing tensor " << tensor_id.str() << " as an IConstantLayer");
+      const_layer->setName(("[Freeze Tensor " + tensor_id.str() + " ]").c_str());
+    } else {
+      // Split converter generates c10::IValue which hold TensorContainer.
+      auto output_container = ptr_.ivalue->toCustomClass<TensorContainer>();
+      out = output_container.get()->tensor();
```

Comment:
This looks good to me. 
Previously for split, we only considered cases with split as the final output or the list unpack as the final output, which is why the bug occurred for ops that followed them.